torchtune/data/_utils.py

```diff
@@ -4,70 +4,15 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-import copy
-from typing import List, Tuple
-
-from torchtune.data._common import CROSS_ENTROPY_IGNORE_IDX
-from torchtune.modules import Tokenizer
-
-
-def tokenize_prompt_and_response(
-    tokenizer: Tokenizer,
-    prompt: str,
-    response: str,
-    train_on_input: bool = False,
-) -> Tuple[List[int], List[int]]:
-    """
-    Tokenize a prompt and response pair.
-
-    Args:
-        tokenizer (Tokenizer): The tokenizer to use.
-        prompt (str): The prompt to tokenize.
-        response (str): The response to tokenize.
-        train_on_input (bool): Whether to train on prompt or mask it out. Default
-            is False.
-
-    Returns:
-        The tokenized prompt and response.
-    """
-    prompt_with_response = prompt + response
-    encoded_prompt = tokenizer.encode(text=prompt, add_bos=True, add_eos=False)
-    encoded_prompt_with_response = tokenizer.encode(
-        text=prompt_with_response, add_bos=True, add_eos=True
-    )
-    labels = copy.deepcopy(encoded_prompt_with_response)
-
-    if not train_on_input:
-        labels[: len(encoded_prompt)] = [CROSS_ENTROPY_IGNORE_IDX] * len(encoded_prompt)
-
-    assert len(encoded_prompt_with_response) == len(labels)
-
-    return encoded_prompt_with_response, labels
+from typing import Any, List
 
 
 def truncate(
-    tokenizer: Tokenizer,
-    prompt_tokens: List[int],
-    label_tokens: List[int],
+    tokens: List[Any],
```

Comment:
Can tokens be anything other than int?

Reply:
So rn I am also calling truncate on mask, which is a list of bools. I am still trying to figure out the cleanest way here.. I do like truncating one sequence at a time better than the existing logic, but the current application to masks is a bit awkward. Lmk if you have any suggestions here